packages/office-ui-fabric-react/src/utilities/decorators/withResponsiveMode.tsx

```diff
@@ -40,74 +51,87 @@ export function setResponsiveMode(responsiveMode: ResponsiveMode | undefined): v
 export function withResponsiveMode<TProps extends { responsiveMode?: ResponsiveMode }, TState>(
   ComposedComponent: new (props: TProps, ...args: any[]) => React.Component<TProps, TState>
 ): any {
-  const resultClass = class WithResponsiveMode extends BaseDecorator<TProps, IWithResponsiveModeState> {
-    constructor(props: TProps) {
-      super(props);
-      this._updateComposedComponentRef = this._updateComposedComponentRef.bind(this);
-
-      this.state = {
-        responsiveMode: _defaultMode || _lastMode || ResponsiveMode.large
-      };
-    }
+  // tslint:disable-next-line:function-name
+  function WithResponsiveModeBase(props: TProps, forwardedRef: React.Ref<React.Component<TProps, TState>>): JSX.Element | null {
+    const [responsiveMode = ResponsiveMode.unknown, setCurrentResponsiveMode] = React.useState<ResponsiveMode>(
+      _defaultMode || _lastMode || ResponsiveMode.large
+    );
 
-    public componentDidMount(): void {
-      this._events.on(window, 'resize', this._onResize);
-      this._onResize();
-    }
+    const rootRef = React.useRef<HTMLDivElement | null>(null);
 
-    public componentWillUnmount(): void {
-      this._events.dispose();
-    }
+    React.useLayoutEffect(() => {
+      if (typeof window !== 'undefined') {
+        const events = new EventGroup(null);
```

Comment:
Do you really need to use EventGroup here?

Instead, why not use add/removeEventListener?

Reply:
We want to deprecate EventGroup over time, now that hooks help with lifetime support.

Additionally we are introducing a `@fluentui/react-event-listener` package which will add `useEventListener` hook and a wrapper `EventListener` component for use in class based components.

But even then, in situations like this, i think just using native eventing works fine.

Reply:
We only add a single event listener to `window` so they're equivalent in terms of risk of memory regression, which is my main concern. I've seen folks commonly forget to keep the two in sync. `EventGroup` is convenient in that regard for its `dispose` method.

Have we marked `EventGroup` as deprecated? We use it extensively across ODSP (sometimes without clean-up).

Reply:
There is tons of usage of `EventGroup` directly across Fabric components. I think a cleaner way to go is to have someone start replacing it for the various scenarios, making it clear what to do in cases like this.